### PR TITLE
fix(docs): include guidance to migrate way from overrides props functions

### LINKED ISSUE #4035: Overrides Props as a function" warning does not provide guidance on what to do.
Ran into the warning which informs that a function passed into the override props is about to be removed in the next version.

For the specific context, I was needing access to the onChange event occurring within the datepickers inner quick select components, e.g. knowing when a quick selection actually used.
https://codesandbox.io/s/base-web-datepicker-override-quickselect-onchange-w9tnv?file=/src/example.js
## Current Behavior
Warning is emitted `baseui:Overrides Props as a function will be removed in the next major version. ` But doesn't say what one can do to fix it.

## Expected Behavior
The warning provides guidance on alternative approaches.

## Your Environment

| Tech    | Version |
| ------- | ------- |
| Base UI | v9.107  |
| React   |         |
| browser |         |

- [x] I have searched the [issues](https://github.com/uber/baseweb/issues) of this repository and believe that this is not a duplicate.

### PR DESCRIPTION
**Pull request by enjoylife**
* Clarify and provide guidance for users seeing the warning: `baseui:Overrides Props as a function will be removed in the next major version`
* Add complex component prop merging example for overrides and datepicker docs.

Fixes #4035

### CODE REVIEW

**Comment by chasestarr** [comment]
Addressed that this feature is buggy and will not be supported in the future. Because of that, we don't want to point anyone towards using it before the feature is removed.

File: src/helpers/overrides.js [hunk 1]
```diff
@@ -110,7 +110,8 @@ export function getOverrides(
     // TODO(v11)
     if (__DEV__) {
       console.warn(
-        'baseui:Overrides Props as a function will be removed in the next major version.',
+        'baseui:Overrides Props as a function will be removed in the next major version. Override the whole component instead. ' +
```
Comment (enjoylife) [inline src/helpers/overrides.js:113]: @chasestarr

**Review by chasestarr — APPROVED** [review]
Thank You

**Comment by chasestarr** [comment]
Can you please run `yarn flow` and `yarn tsc` to typecheck the new example?

**Comment by enjoylife** [comment]
fixed lint

**Pull request merged**